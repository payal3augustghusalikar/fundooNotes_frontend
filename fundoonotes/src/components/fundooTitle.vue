
<template>

  <v-card-title class="md-title">
     <span class="text-family-size" id="col">F</span>
    <span class="md-title" id="col1">u</span>
    <span class="md-title" id="col2">n</span>
    <span class="md-title" id="col3">d</span>
    <span class="md-title" id="col4">o</span>
    <span class="md-title" id="col5">o</span>
    <span class="md-title" id="col6">N</span>
    <span class="md-title" id="col7">o</span>
    <span class="md-title" id="col8">t</span>
    <span class="md-title" id="col9">e</span>
    <span class="md-title" id="col10">s</span>
 
  </v-card-title>
   
</template>

<script>


export default {

};
</script>
<style lang="scss" scoped>

// @import url("../scss/fundooTitle.scss");
@import "../scss/fundooTitle.scss"
</style>
