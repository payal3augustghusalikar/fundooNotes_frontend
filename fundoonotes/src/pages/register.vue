<template>
  <div>

  
    <form novalidate class="md-layout jc-center" @submit.prevent="validateUser">

      <md-card class="md-layout-item md-size-50 md-small-size-100 overflow-x">

         
        <md-card-header>
        
      <div class="md-title" v-bind:style="styleObject" >
     
      <v-card-title>
  </v-card-title>
    <fundooTitle />
      </div>
     
      <div class="md-title">Create your Account</div>
        </md-card-header>
 
        <md-card-content>
          <div class="md-layout md-gutter">
            <div class="md-layout-item md-small-size-100">
              <md-field :class="getValidationClass('firstName')">
                <label for="first-name">First Name</label>
                <md-input-box name="first-name" id="first-name" autocomplete="given-name" v-model="form.firstName" :disabled="sending" />
                <span class="md-error" v-if="!$v.form.firstName.required">The first name is required</span>
                <span class="md-error" v-else-if="!$v.form.firstName.minlength">Invalid first name</span>
              </md-field>
            </div>

            <div class="md-layout-item md-small-size-100">
              <md-field :class="getValidationClass('lastName')">
                <label for="last-name">Last Name</label>
                <md-input name="last-name" id="last-name" autocomplete="family-name" v-model="form.lastName" :disabled="sending" />
                <span class="md-error" v-if="!$v.form.lastName.required">The last name is required</span>
                <span class="md-error" v-else-if="!$v.form.lastName.minlength">Invalid last name</span>
              </md-field>
            </div>
          </div>

        <!--  <md-field :class="getValidationClass('email')">
            <label for="phone">Phone Number</label>
            <md-input  outlined dense type="number" name="phone" id="phone" autocomplete="phone" v-model="form.phone" :disabled="sending" />
            <span class="md-error" v-if="!$v.form.phone.required">The phone number is required</span>
            <span class="md-error" v-else-if="!$v.form.phone.maxLength">Invalid Phone Number</span>
          </md-field>-->

          <md-field :class="getValidationClass('email')">
            <label for="email">Email</label>
            <md-input type="email" name="email" id="email" autocomplete="email" v-model="form.email" :disabled="sending" />
            <span class="md-error" v-if="!$v.form.email.required">The email is required</span>
            <span class="md-error" v-else-if="!$v.form.email.email">Invalid email</span>
          </md-field>

          <div class="md-layout md-gutter">
            <div class="md-layout-item md-small-size-100">
              <md-field :class="getValidationClass('password')">
                <label for="password">Password</label>
                <md-input name="password" type="password" id="password" v-model="form.password" :disabled="sending" />
                <span class="md-error" v-if="!$v.form.password.required">The password is required</span>
                <span class="md-error" v-else-if="!$v.form.password.minlength">Password should contain minimum 4 charecters </span>
              </md-field>
            </div>

            <div class="md-layout-item md-small-size-100">
              <md-field :class="getValidationClass('password')">
                <label for="cpassword">Confirm Password</label>
                <md-input name="cpassword" type="password" id="cpassword" v-model="form.cpassword" :disabled="sending" />
                <span class="md-error" v-if="!$v.form.cpassword.required">The confirm password is required</span>
                <span class="md-error" v-else-if="!$v.form.cpassword.minlength">Confirm Password should contain minimum 4 charecters </span>
              </md-field>
            </div>
          </div>

          
        </md-card-content>

        <md-progress-bar md-mode="indeterminate" v-if="sending" />

<md-card-content>
        <md-card-actions>
           <md-button type="submit" class="md-primary" :disabled="sending" >sign in instead </md-button>
          <md-button type="submit" class="md-raised md-primary" :disabled="sending">Next</md-button>
       
        </md-card-actions>
</md-card-content> 

       </md-card>
      <figure class="OFqWT"><img src="https://ssl.gstatic.com/accounts/signup/glif/account.svg" alt="" width="244" height="244" class="j9NuTc TrZEUc"><figcaption class="oEvHdd">One account. All of Google working for you.</figcaption></figure>
 
      <md-snackbar :md-active.sync="userSaved">The user {{ lastUser }} was saved with success!</md-snackbar>  
    </form>
  </div>
</template>

<script>
import fundooTitle from '../components/fundooTitle.vue';
//import title from "../../components/title.vue";
  import { validationMixin } from 'vuelidate'
 
  import {
    required,
    email,
    minLength,
    // maxLength,
    // password
  } from 'vuelidate/lib/validators'
 // import userApi from '@/services/api/user'
  export default {


 components: {
 fundooTitle
  },



    name: 'signUp',
    mixins: [validationMixin],
    data: () => ({
      form: {
        firstName: null,
        lastName: null,
        email: null,
        password: null,
        cpassword: null,
      },
      
      userSaved: false,
      sending: false,
      lastUser: null
    }),

    validations: {
      form: {
        firstName: {
          required,
          minLength: minLength(3)
        },
        lastName: {
          required,
          minLength: minLength(3)
        },
         email: {
          required,
          email
        },
        //  phone: {
        //   required,
        //    minLength: minLength(10)
        // },
        password: {
          required,
          minLength: minLength(4)
        },
        cpassword: {
          required,
          minLength: minLength(4)
        },
       
      }
    },
  
   
  

    methods: {
    
      getValidationClass (fieldName) {
        const field = this.$v.form[fieldName]
        if (field) {
          return {
            'md-invalid': field.$invalid && field.$dirty
          }
        }
      },
      clearForm () {
        this.$v.$reset()
        this.form.firstName = null
        this.form.lastName = null
        this.form.email = null
        this.form.phone = null
        this.form.password = null
        this.form.cpassword = null 
      
      },
      saveUser () {
        this.sending = true
        let data = {
            firstName: this.form.firstName,
            lastName: this.form.lastName,
            phone: this.form.phone,
            email: this.form.email,
            password: this.form.password
   
   }
        // Instead of this timeout, here you can call your API
        console.log("signup details: ",data)
     //   userApi.registerUser(data)
        .then(result => {
          console.log("Success", result)
          window.setTimeout(() => {
            this.lastUser = `${data.firstName} ${data.lastName}`
            this.userSaved = true
            this.sending = false
            this.clearForm()
          }, 1500)
        })
        .catch(error => console.log(error))
      },
      validateUser () {
        this.$v.$touch()
        if (!this.$v.$invalid) {
          this.saveUser()
        }
      }
    }
  }
</script>

<style scoped>

@import url("../scss/register.scss");
    
</style>
