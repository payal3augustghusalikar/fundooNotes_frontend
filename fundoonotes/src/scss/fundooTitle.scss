 .md-title #col {
     color: blue;
 }
 
 .md-title #col1 {
     color: red;
 }
 
 .md-title #col2 {
     color: orange;
 }
 
 .md-title #col3 {
     color: blue;
 }
 
 .md-title #col4 {
     color: green;
 }
 
 .md-title #col5 {
     color: red;
 }
 
 .md-title #col6 {
     color: blue;
 }
 
 .md-title #col7 {
     color: red;
 }
 
 .md-title #col8 {
     color: orange;
 }
 
 .md-title #col9 {
     color: blue;
 }
 
 .md-title #col10 {
     color: green;
 }
 
 