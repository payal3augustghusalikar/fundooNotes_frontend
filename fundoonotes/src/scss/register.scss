.md-progress-bar {
    position: absolute;
    top: 0;
    right: 0;
    left: 0;
}

.md-title,
.card {
    padding-top: 0px;
    padding-bottom: 0px;
}

.password-hint {
    margin-top: -25px;
    font-size: 11px;
    justify-content: left;
    color: rgb(109, 107, 107);
    margin-top: -25px;
    position: absolute;
    margin-left: 0px;
    font-size: 11px;
}

.account-img {
    margin-left: 0px;
    margin-top: 70px;
    padding-left: 0px;
    padding-right: 0px;
}

.md-layout-item {
    padding-top: 0px
}

.md-layout {
    justify-self: center;
    padding-top: 0px;
}

.jc-center {
    justify-content: center;
}

.overflow-x {
    overflow-x: hidden
}

.md-card {
    padding-top: 0px;
}

::-webkit-scrollbar {
    display: none;
}

.register {
    padding-top: 100px;
    padding-right: 10 px;
}