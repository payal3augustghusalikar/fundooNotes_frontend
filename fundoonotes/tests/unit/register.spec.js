import { createLocalVue, shallowMount } from '@vue/test-utils';
import fundooTitle from '../../src/components/fundooTitle.vue';
import register from '../../src/pages/register.vue';
import Vuelidate from 'vuelidate';
import Vuetify from 'vuetify';
import VueMaterial from 'vue-material';

describe('register.vue', () => {
    let wrapper;
    beforeEach(() => {
        const localVue = createLocalVue();
        localVue.use(Vuetify);
        localVue.use(VueMaterial);
        localVue.use(Vuelidate);
        wrapper = shallowMount(register, localVue);
    });

    it('givenPage_ifFundooTitleIsPresent_shouldReturnTrue', () => {
        expect(wrapper.contains(fundooTitle)).toBe(true);
    });

    it('has data', () => {
        expect(typeof register.data).toBe('function');
    });

    it('has a button', () => {
        expect(wrapper.contains('md-button')).toBe(true);
    });

    it('html should render correctly', () => {
        expect(wrapper.html()).toMatchSnapshot()
    })

    it('inGivenFirstName_whenNotPresent_shouldEvaluateFalse', () => {
        wrapper.vm.form.firstName = '';
        expect(wrapper.vm.$v.form.firstName.required).toBe(false);
    });

    it('inGivenFirstName_whenGivenMoreThan3char_shouldEvaluate"$invalid"False', () => {
        wrapper.vm.form.firstName = 'Payal';
        expect(wrapper.vm.$v.form.firstName.$invalid).toBe(false);
    });

    it('inGivenFirstName_whenGivenLessThan3char_shouldEvaluate"$invalid"True', () => {
        wrapper.vm.form.firstName = 'Pa';
        expect(wrapper.vm.$v.form.firstName.$invalid).toBe(true);
    });

    it('inGivenLastName_whenNotPresent_shouldEvaluateFalse', () => {
        wrapper.vm.form.lastName = '';
        expect(wrapper.vm.$v.form.lastName.required).toBe(false);
    });

    it('inGivenLastName_whenGivenMoreThan3char_shouldEvaluate"$invalid"False', () => {
        wrapper.vm.form.lastName = 'Payal';
        expect(wrapper.vm.$v.form.lastName.$invalid).toBe(false);
    });

    it('inGivenLastName_whenGivenLessThan3char_shouldEvaluate"$invalid"True', () => {
        wrapper.vm.form.lastName = 'Pa';
        expect(wrapper.vm.$v.form.lastName.$invalid).toBe(true);
    });



    it('inGivenEmail_whenProper_shouldEvaluateTrue', () => {
        wrapper.vm.form.email = 'payalghusalikar9655@gmail.com';
        expect(wrapper.vm.$v.form.email.email).toBe(true);
    });

    it('givenEmailId_ifEmpty_shouldEvaluateFalse', () => {
        wrapper.vm.form.email = '';
        expect(wrapper.vm.$v.form.email.required).toBe(false);
    });

    it('inGivenEmail_whenImproper_shouldEvaluateFalse', () => {
        wrapper.vm.form.email = ',,.,.,.@gmail.com';
        expect(wrapper.vm.$v.form.email.email).toBe(false);
    });

    it('inGivenEmail_when"@"isNotPresent_shouldEvaluateFalse', () => {
        wrapper.vm.form.email = 'payalghusalikar9gmail.com';
        expect(wrapper.vm.$v.form.email.email).toBe(false);
    });

    it('inGivenEmail_when".com"NotPresent_shouldEvaluateFalse', () => {
        wrapper.vm.form.email = 'payalghusalikar9gmail';
        expect(wrapper.vm.$v.form.email.email).toBe(false);
    });

    it('inGivenEmail_whenImproper_shouldEvaluateFalse', () => {
        wrapper.vm.form.email = 'pl123&&&';
        expect(wrapper.vm.$v.form.email.email).toBe(false);
    });
    it('inGivenPassword_whenNotPresent_shouldEvaluateFalse', () => {
        wrapper.vm.form.password = '';
        expect(wrapper.vm.$v.form.password.required).toBe(false);
    });

    it('inGivenPassword_whenGivenMoreThan6char_shouldEvaluate"$invalid"False', () => {
        wrapper.vm.form.password = 'AAAAAAAA';
        expect(wrapper.vm.$v.form.password.$invalid).toBe(false);
    });

    it('inGivenPassword_whenGivenLessThan3char_shouldEvaluate"$invalid"True', () => {
        wrapper.vm.form.password = 'AAAAA';
        expect(wrapper.vm.$v.form.password.$invalid).toBe(true);
    });

    it('inGivenConfirmPassword_whenNotPresent_shouldEvaluateFalse', () => {
        wrapper.vm.form.cpassword = '';
        expect(wrapper.vm.$v.form.cpassword.required).toBe(false);
    });

    it('inGivenConfirmPassword_whenGivenMoreThan6char_shouldEvaluate"$invalid"False', () => {
        wrapper.vm.form.cpassword = 'AAAAAAAA';
        expect(wrapper.vm.$v.form.cpassword.$invalid).toBe(false);
    });

    it('inGivenConfirmPassword_whenGivenLessThan3char_shouldEvaluate"$invalid"True', () => {
        wrapper.vm.form.cpassword = 'AAAAA';
        expect(wrapper.vm.$v.form.cpassword.$invalid).toBe(true);
    });

    // it('inGivenlastName_whenNotPresent_shouldEvaluateFalse', () => {
    //     wrapper.vm.form.lastName = '';
    //     expect(wrapper.vm.$v.form.lastName.required).toBe(true);
    // });

    // it('inGivenpassword_whenNotPresent_shouldEvaluateFalse', () => {
    //     wrapper.vm.password = '';
    //     expect(wrapper.vm.$v.password.required).toBe(false);
    // });

    // it('inGivenpassword_whenNotPresent_shouldEvaluateFalse', () => {
    //     wrapper.vm.password = '';
    //     expect(wrapper.vm.$v.password.invalid).toBe(true);
    // });

    // it('inGivenCpassword_whenNotPresent_shouldEvaluateFalse', () => {
    //     wrapper.vm.Cpassword = '';
    //     expect(wrapper.vm.$v.Cpassword.required).toBe(false);
    // });

    // it('inGivenCpassword_whenNotPresent_shouldEvaluateFalse', () => {
    //     wrapper.vm.Cpassword = '';
    //     expect(wrapper.vm.$v.Cpassword.invalid).toBe(true);
    // });



});