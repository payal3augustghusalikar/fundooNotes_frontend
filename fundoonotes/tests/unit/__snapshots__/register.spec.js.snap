// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`register.vue html should render correctly 1`] = `
<div>
  <form novalidate="novalidate" class="md-layout jc-center register">
    <md-card class="md-layout-item md-size-50 md-small-size-120 overflow-y overflow-x ">
      <div class="md-layout md-gutter">
        <div class="md-layout-item md-small-size-100">
          <md-card-header>
            <md-card-title>
              <fundootitle-stub></fundootitle-stub>
            </md-card-title>
            <v-card-title>
              Create your Account
            </v-card-title>
          </md-card-header>
          <md-card-content>
            <div class="md-layout md-gutter">
              <div class="md-layout-item md-small-size-100">
                <md-field class=""><label for="first-name">First Name</label>
                  <md-input autocomplete="off" label="First name"></md-input> <span class="md-error">The first name is required</span>
                </md-field>
              </div>
              <div class="md-layout-item md-small-size-50">
                <md-field class=""><label for="last-name">Last Name</label>
                  <md-input name="last-name" autocomplete="off" outline="" dense=""></md-input> <span class="md-error">The last name is required</span>
                </md-field>
              </div>
            </div>
            <md-field class=""><label for="email">Email</label>
              <md-input type="email" name="email" autocomplete="off"></md-input> <span class="md-error">The email is required</span>
            </md-field>
            <div class="md-layout md-gutter">
              <div class="md-layout-item md-small-size-100">
                <md-field class=""><label for="password">Password</label>
                  <md-input name="password" type="password"></md-input> <span class="md-error">The password is required</span>
                </md-field>
              </div>
              <div class="md-layout-item md-small-size-100">
                <md-field class=""><label for="cpassword">Confirm</label>
                  <md-input name="cpassword" type="password"></md-input> <span class="md-error">The confirm password is required</span>
                </md-field>
              </div>
            </div>
            <p class="password-hint">
              Use 8 or more characters with a mix of letters, numbers &amp;
              symbols
            </p>
          </md-card-content>
          <!---->
          <md-card-content>
            <md-card-actions><span><router-link-stub to="/login" tag="a" ariacurrentvalue="page" event="click">sign in instead</router-link-stub></span>
              <v-spacer></v-spacer>
              <md-button type="submit" class="md-raised md-primary">Next</md-button>
            </md-card-actions>
          </md-card-content>
        </div>
        <div class="md-layout md-small-size-50">
          <figure class="account-img"><img src="https://ssl.gstatic.com/accounts/signup/glif/account.svg" alt="" width="244" height="244" class="j9NuTc TrZEUc">
            <figcaption class="oEvHdd">
              One account. All of Google working for you.
            </figcaption>
          </figure>
        </div>
      </div>
      <md-snackbar>The user was saved with success please
        login!</md-snackbar>
      <md-snackbar>Error Occured please try again!</md-snackbar>
    </md-card>
  </form>
</div>
`;
